src/core_modules/capture-ui/DateAndTimeFields/DateField/Date.component.js

```diff
@@ -1,206 +1,76 @@
+    calendar?: string,
 };
 
-type State = {
-    popoverOpen: boolean,
-};
-
-export class DateField extends React.Component<Props, State> {
-    static splitPassOnProps(passOnProps: ?Object) {
-        const splittedProps = {
-            input: {},
-            popup: {},
-            calendar: {},
-        };
-
-        if (!passOnProps) {
-            return splittedProps;
-        }
-
-        return Object
-            .keys(passOnProps)
-            .reduce((accSplittedProps, propKey) => {
-                let propContainer;
-                if (propKey.startsWith(DateField.propContainers.CALENDAR)) {
-                    propContainer = DateField.propContainers.CALENDAR;
-                } else if (propKey.startsWith(DateField.propContainers.POPUP)) {
-                    propContainer = DateField.propContainers.POPUP;
-                } else {
-                    propContainer = DateField.propContainers.INPUT;
-                }
-
-                const outputKey = lowerCaseFirstLetter(propKey.replace(propContainer, ''));
-                accSplittedProps[propContainer][outputKey] = passOnProps[propKey];
-                return accSplittedProps;
-            }, splittedProps);
-    }
-
-    containerInstance: ?HTMLElement;
-    handleTextFieldFocus: () => void;
-    handleDateSelected: (value: string) => void;
-    handleTextFieldBlur: (event: SyntheticEvent<HTMLInputElement | HTMLTextAreaElement>) => void;
-    hidePopover: () => void;
-    handleDocumentClick: (event: MouseEvent) => void;
-    calendarWrapperDOMElementRef: { current: ?HTMLDivElement };
+export class DateField extends React.Component<Props> {
+    handleDateSelected: (value: {calendarDateString: string}) => void;
 
     constructor(props: Props) {
         super(props);
 
-        this.state = {
-            popoverOpen: false,
-        };
-
-        this.handleTextFieldFocus = this.handleTextFieldFocus.bind(this);
         this.handleDateSelected = this.handleDateSelected.bind(this);
-        this.handleTextFieldBlur = this.handleTextFieldBlur.bind(this);
-        this.hidePopover = this.hidePopover.bind(this);
-        this.handleDocumentClick = this.handleDocumentClick.bind(this);
-
-        this.calendarWrapperDOMElementRef = createRef();
-    }
-
-    componentWillUnmount() {
-        document.removeEventListener('click', this.handleDocumentClick);
-    }
-
-    static propContainers = {
-        CALENDAR: 'calendar',
-        POPUP: 'popup',
-        INPUT: 'input',
-    };
-
-    handleTextFieldFocus() {
-        document.removeEventListener('click', this.handleDocumentClick);
-
-        this.setState({
-            popoverOpen: true,
-        });
-
-        this.props.onFocus && this.props.onFocus();
     }
 
-    handleDateSelected(value: string) {
-        this.props.onBlur(value);
-        this.hidePopover();
+    handleDateSelected(value: {calendarDateString: string}) {
+        const date = value?.calendarDateString;
+        this.props.onBlur(date);
```

Comment:
When the date is mandatory, the red error background appears for a second. 

https://github.com/user-attachments/assets/1afd5dcc-88c5-4073-9618-a8714ae9e8a2

I think that checking for `undefined` will fix this.
```
if (date !== undefined) {
    this.props.onBlur(date);
}
```
WDYT? Thanks!

Reply:
Yes! it solved the issue. Thanks